File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}

        for i in nums:
            counts[i] = counts.get(i, 0) + 1
        
        freq = [[] for _ in range(len(nums)+1)]
        print(freq)
        for i,j in counts.items():
            freq[j].append(i)
        res = []
        for i in range(len(nums), 0, -1):
            for j in freq[i]:
                res.append(j)
                if len(res) == k:
                    return res

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] a = new int[nums.length];
        Arrays.fill(a, 1);
        int c = 1;
        for(int i = 0;i<nums.length;i++){a[i] *= c; c *=nums[i]; }
        c = 1;
        for(int i = nums.length - 1;i>-1;i--){a[i] *= c; c *=nums[i]; }
        return a;
    }

}  
